processing/src/test/java/org/apache/druid/timeline/VersionedIntervalTimelineTest.java

```diff
@@ -470,6 +470,66 @@ public void testOverlapSecondContainsFirst()
     );
   }
 
+  //  1|----|----|
+  //  2|---------|
+  @Test
+  public void testOverlapSecondContainsFirstZeroLengthInterval()
+  {
+    add("2011-01-01/2011-01-20", "1", 1);
+    add("2011-01-01/2011-01-10", "2", 2);
+    add("2011-01-10/2011-01-20", "2", 3);
+
+    assertValues(
+        Collections.emptyList(),
+        timeline.lookup(Intervals.of("2011-01-10/2011-01-10"))
+    );
+  }
+
+  //  1|----|----|
+  //  2|---------|
+  @Test
+  public void testOverlapSecondContainsFirstZeroLengthInterval2()
+  {
+    add("2011-01-01/2011-01-20", "1", 1);
+    add("2011-01-01/2011-01-10", "2", 2);
+    add("2011-01-10/2011-01-20", "2", 3);
+
+    assertValues(
+        Collections.emptyList(),
+        timeline.lookup(Intervals.of("2011-01-11/2011-01-11"))
+    );
+  }
+
+  //  1|----|----|
+  //  2|---------|
+  @Test
+  public void testOverlapSecondContainsFirstZeroLengthInterval3()
+  {
+    add("2011-01-01/2011-01-20", "1", 1);
+    add("2011-01-01/2011-01-10", "2", 2);
+    add("2011-01-10/2011-01-20", "2", 3);
+
+    assertValues(
+        Collections.emptyList(),
+        timeline.lookup(Intervals.of("2011-01-01/2011-01-01"))
+    );
+  }
+
+  //  1|----|----|
+  //  2|---------|
+  @Test
+  public void testOverlapSecondContainsFirstZeroLengthInterval4()
```

Comment:
Are these tests meant to be duplicates of each other?

Reply:
They are slightly different (check the arguments to `timeline.lookup`)